src/client_side.cc

```diff
@@ -3058,11 +3058,10 @@ ConnStateData::splice()
 {
     // normally we can splice here, because we just got client hello message
 
-    if (fd_table[clientConnection->fd].ssl.get()) {
-        // Restore default read methods
-        fd_table[clientConnection->fd].read_method = &default_read_method;
-        fd_table[clientConnection->fd].write_method = &default_write_method;
-    }
+    // fde::ssl/tls_read_method() probably reads from our own inBuf. If so, then
```

Comment:
The uncertainty here worries me a bit. Please at least add a TODO for someone else to follow up on which states what needs to be done to get surety.

Reply:
I share your worries, but, FWIW, I cannot formulate a useful TODO here. IMO, the current comment (including its intended uncertainty) reflects the current (shallow) understanding of this code well. Anybody with enough expertise to resolve this uncertainty ought to be able to do so without an explicit TODO.